app/main.py

```diff
@@ -1 +1,32 @@
-# write your code here
+class Animal:
+    health = 100
+    name = ""
+    hidden = False
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.alive.append(self)
```

Comment:
```suggestion
    alive = []

    def __init__(
            self,
            name: str,
            health: int = 100,
            hidden: bool = False
    ) -> None:
        self.name = name
        self.health = health
        self.hidden = hidden
        Animal.alive.append(self)
```
Ти задаєш атрибути health, name для класу Animal, але при створенні нового екземпляру, йому буде надаватися дефолтне значення з `__init__` для health, а також тварина завжди буде мати ім’я, тож ці параметри не будуть наслідуватись. Напевно краще перенести все в init, а ці рядки видалити, залишивши тільки список alive. А також, краще додавати референс відразу в список Animal.alive, а не через екземпляр self.

Reply:
дякую!!